pages/courses/index.js

```diff
@@ -18,18 +19,45 @@ export default function Courses({ courses }) {
     </Head>
   )
 
+  const perPage = 100
+  const [courses, setCourses] = useState(allCourses.slice(0, perPage))
```

Comment:
May be just a tiny optimization?

```suggestion
  const [courses, setCourses] = useState(() => allCourses.slice(0, perPage))
```

https://beta.reactjs.org/reference/react/useState#avoiding-recreating-the-initial-state